pkg/daemon/daemon.go

```diff
@@ -931,3 +940,74 @@ func tryCreateNMUdevRule(unsupportedNicIdMap map[string]string) error {
 	}
 	return nil
 }
+
+func (dn *Daemon) tryMellanoxCheck() error {
+	glog.Info("tryMellanoxCheck(): ")
+
+	var interfaces sriovnetworkv1.InterfaceExts
+OUTER:
+	for _, ifaceStatus := range dn.nodeState.Status.Interfaces {
+		if ifaceStatus.Vendor != MellanoxVendorId {
+			continue
+		}
+
+		if ifaceStatus.NumVfs == 0 {
+			continue
+		}
+		if ifaceStatus.Name == "" {
+			continue
+		}
+		for _, ifaceSpec := range dn.nodeState.Spec.Interfaces {
+			if ifaceSpec.PciAddress != ifaceStatus.PciAddress {
+				continue
+			}
+			// Skip non RDMA NICs
+			if !ifaceSpec.IsRdma {
+				continue
+			}
+
+			pfLink, err := netlink.LinkByName(ifaceStatus.Name)
+			if err != nil {
+				return err
+			}
+			for _, vf := range pfLink.Attrs().Vfs {
+				if vf.Mac.String() == "00:00:00:00:00:00" {
```

Comment:
you should check if the node guid is zero
[root@l-dcs87 ~]# cat /sys/class/net/enp3s1f3/device/infiniband/mlx5_5/node_guid 
0000:0000:0000:0000